common/lib/xmodule/xmodule/modulestore/split_mongo/split.py

```diff
@@ -1128,14 +1128,36 @@ def get_parent_location(self, locator, **kwargs):
 
         :param locator: BlockUsageLocator restricting search scope
         """
+        def is_root_block(xblock, is_root=False):
+            """ Check recursively if an xblock has a path to the course root """
+            xblock_parents = self._get_parents_from_structure(xblock, course.structure)
+
+            if len(xblock_parents) == 0 and xblock.type == "course":
+                # Found, xblock has the path to the root
+                return True
+
+            for xblock_parent in xblock_parents:
+                is_root = is_root_block(xblock_parent, is_root)
```

Comment:
This would just be simpler to write as
```
return any(is_root_block(xblock_parent) for xblock_parent in xblock_parents)
```
Then you avoid unnecessary iterations if it turns out that one of the parents has a path to the root.